recipes/grpc/all/conanfile.py

```diff
@@ -95,35 +101,32 @@ def requirements(self):
         self.requires("c-ares/1.18.1")
         self.requires("openssl/1.1.1q")
         self.requires("re2/20220601")
-        self.requires("zlib/1.2.12")
+        self.requires("zlib/1.2.13")
         self.requires("protobuf/3.21.4")
```

Comment:
When cross compiling, shoulnd't the protoc binaries only be built for the host platform and not the target platform?

```python
self.options["protobuf"].with_protoc_binaries = False
```

However, this will only work if protobuf recipe is updated https://github.com/fjp/conan-center-index/commit/cc59b5362fea4cee197614aa921058d72cdc5709

Reply:
it doesn't matter

Reply:
Thanks for your quick response.

Sure, it can also be set in a profile or from the command line. But my point is that there seems to be an issue when cross building grpc. Cross-build from 'Linux:x86_64' to 'Linux:armv7hf' using `-b missing` I get build errors for protobuf:

```
...
usr/include/c++/v1/cstdlib:123:9: error: target of using declaration conflicts with declaration already in scope
using ::abs _LIBCPP_USING_IF_EXISTS;
...
protobuf/3.20.0: ERROR: Package '4c113c49faf2c4384d5b6d4b12ce3565050d800e' build failed
protobuf/3.20.0: WARN: Build folder /root/.conan/data/protobuf/3.20.0/_/_/build/4c113c49faf2c4384d5b6d4b12ce3565050d800e
root@0ec9a559fe37:~# conan or in build() method, line 203

        ConanException: Error 2 while executing cmake --build '/root/.conan/data/protobuf/3.20.0/_/_/build/4c113c49faf2c4384d5b6d4b12ce3565050d800e/build_subfolder' '--' '-j8'
```

Are there any build jobs on conan-center-index that run cross compilation? I have just seen https://github.com/conan-io/conan-center-index/pull/13817#issuecomment-1294929455 which doesn't seem to use different host and build profiles, but please correct me if I am wrong.

Reply:
In c3i, there is one cross-build with 2 profiles: from macOS Intel to macOS M1.

From what I see from your log, it's an issue for protobuf recipe when target is Linux armv7hf.